src/backend/src/lib.rs

```diff
@@ -179,6 +182,17 @@ pub fn config() -> Config {
     read_config(std::clone::Clone::clone)
 }
 
+/// Adds cycles to the cycles ledger, if it is below a certain threshold.
+///
+/// # Errors
+/// Error conditions are enumerated by: `TopUpCyclesLedgerError`
+#[update(guard = "caller_is_allowed")]
+pub async fn top_up_cycles_ledger(
```

Comment:
Done.  And moved the types to shared/types, which looks like a better place for them.

I cannot write the full test suite until I have a test env with the ledger and ideally the signer, though, which is why the full test suite will come later.

Reply:
Absolutely.  Also, the other tests are held up because they need other canisters to be set up, whereas this test can be done with just the backend canister so doesn't need to wait.  So good point!

Reply:
Oh.  Formatting.  Push should work now.

Reply:
Test is in it/signer.rs